src/haversine.jl

```diff
@@ -0,0 +1,42 @@
+"""
+    Haversine(radius)
+
+The haversine distance between two locations on a sphere of given `radius`.
+
+Locations are described with longitude and latitude in degrees and
+the radius of the Earth is used by default (≈ 6371km). The computed
+distance has the same units as that of the radius.
+
+### Notes
+
+The haversine formula is widely used to approximate the geodesic distance
+between two points at the surface of the Earth. The error from approximating
+the Earth as a sphere is typically negligible for most applications. It is
+no more than 0.3%.
+"""
+struct Haversine{T<:Real} <: Metric
+  radius::T
+end
+
+# use Earth radius ≈ 6371km by default
+Haversine() = Haversine(6371.)
```

Comment:
I really don't think we should define this here. As with the note in the docstring, it's too specific for such a general package, and in this case I think it's highly error-prone.

Reply:
@ararslan I think by far the application of the Haversine is related to measurements on our planet, few people will use it with a different radius. If it is necessary to remove the default radius, I can do it, I just don't see this as a restriction, it is a default value only.

Reply:
Elaborating on this, when people talk about this distance they are usually thinking about geosciences. I personally don't think we are missing any generality here as the user can still use any radius he/she wants.